src/main.cc

```diff
@@ -620,41 +620,117 @@ print_version(void)
-		"Run interactive LuaJIT interpreter:\n\n"
-		"  %s -i\n\n"
-		"Usage:\n\n"
-		"  %s [OPTIONS] [SCRIPT [ARGS]]\n\n"
-		"Options:\n\n"
-		" -h, --help             display this help and exit\n"
-		" --help-env-list        display env variables taken into account\n"
-		" --force-recovery       enable force-recovery\n"
-		" -v, --version          print program version and exit\n"
-		" -c, --config PATH      set a path to yaml config file as 'PATH'\n"
-		" -n, --name INSTANCE    set an instance name as 'INSTANCE'\n"
-		" -i                     enter interactive mode\n"
-		" -e EXPR                execute string 'EXPR'\n"
-		" -l NAME                require library 'NAME'\n"
-		" -j cmd                 perform LuaJIT control command\n"
-		" -b ...                 save or list bytecode\n"
-		" -d                     activate debugging session for script\n"
-		" --failover             run a failover coordinator (Enterprise Edition)\n"
-		" --                     stop handling options\n"
-		" -                      execute stdin and stop handling options\n"
-		"\n"
-		"Please visit project home page at https://tarantool.org\n"
-		"to see online documentation, submit bugs or contribute a patch.\n"
-		;
-	fprintf(stream, help_msg, tarantool_version(),
-		program, program, program, program);
+	static const char help_msg[] = "Tarantool %s\n"
+		"\n"
+		"----------------------------------------------------------\n"
+		"\n"
+		"The recommended way to work with Tarantool is to use tt\n"
+		"tool.\n"
+		"\n"
+		"Run a Tarantool instance:\n"
+		"\n"
+		"$ tt start --name <instance name>\n"
+		"\n"
+		"Connect to an instance:\n"
+		"\n"
+		"$ tt connect <uri>\n"
+		"\n"
+		"Execute a Lua script:\n"
+		"\n"
+		"$ tt run [OPTIONS] script.lua [ARGS]\n"
+		"\n"
+		"Run an interactive Lua interpreter:\n"
+		"\n"
+		"$ tt run -i\n"
+		"\n"
+		"----------------------------------------------------------\n"
+		"\n"
+		"At the lower level there are the following action options.\n"
+		"\n"
+		"-h\n"
+		"--help\n"
+		"\n"
+		"    Print this help message.\n"
+		"\n"
+		"-v\n"
+		"--version\n"
+		"\n"
+		"    Print version and build information.\n"
+		"\n"
+		"--help-env-list\n"
+		"\n"
+		"    Print environment variables list.\n"
+		"\n"
+		"-n <...> [-c <...>]\n"
+		"--name <...> [--config <...>]\n"
+		"\n"
+		"    Start an instance.\n"
+		"\n"
+		"    The path to the configuration file can be omitted if a\n"
+		"    configuration is stored in etcd and TT_CONFIG_ETCD_*\n"
+		"    environment variables are set.\n"
+		"\n"
+		"<first positional argument> [<..more args..>]\n"
+		"\n"
+		"    Run a Lua script file. All the arguments after the\n"
+		"    script name are stored in the `arg` global value.\n"
+		"\n"
+		"    The file may be pointed as `-` to read the script from\n"
+		"    stdin.\n"
+		"\n"
+		"-e <..code..>\n"
+		"\n"
+		"    Run a Lua code.\n"
+		"\n"
+		"    This option can be used together with other action\n"
+		"    options or on its own.\n"
```

Comment:
Minor: It's worth to mention, that `-e` can be also used as a *modifier*. Feel free to ignore.

Reply:
I'll clarify it a bit.

```diff
diff --git a/src/main.cc b/src/main.cc
index ae612baca1..33818950c7 100644
--- a/src/main.cc
+++ b/src/main.cc
@@ -684,8 +684,8 @@ print_help(FILE *stream)
                "\n"
                "    Run a Lua code.\n"
                "\n"
-               "    This option can be used together with other action\n"
-               "    options or on its own.\n"
+               "    It can be used on its own (as an action option) or\n"
+               "    together with another action option (as a modifier).\n"
                "\n"
                "-i\n"
                "\n"
```

Reply:
Done.